### PR TITLE
Introduce CI to run a linter and formatter

### PR DESCRIPTION
**Pull request by gurrpi**
Changes:
* Add initial CI that runs linter and formatter 
* Relocates ESLint and Prettier to root project scope for consistent use
* Apply Prettier to `contracts` package
* Enforce double quotes to solidity files

You can apply Prettier formatting by running `yarn run prettier:fix` in repository root.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,15 +1,12 @@
 {
-  "name": "@qfi/qfi",
+  "name": "@quadratic-funding/qfi",
```
Comment (daodesigner) [inline package.json:2]: do not change package names

File: package.json [hunk 2]
```diff
@@ -22,13 +19,27 @@
...
-    "build:subgraph": "yarn workspace @qfi/subgraph run build",
-    "build:ui": "yarn workspace @qfi/ui run build",
-    "build:hooks": "yarn workspace @qfi/hooks run build",
+    "build:contracts": "yarn workspace @quadratic-funding/contracts run build",
+    "build:subgraph": "yarn workspace @quadratic-funding/subgraph run build",
+    "build:ui": "yarn workspace @quadratic-funding/ui run build",
+    "build:hooks": "yarn workspace @quadratic-funding/hooks run build",
+    "lint": "npm-run-all lint:*",
+    "lint:eslint": "eslint packages",
+    "lint:prettier": "prettier --check **/*.ts",
```
Comment (daodesigner) [inline package.json:28]: Nit: remove, do not add multiple linters to packages, follow the conventions each subpackage uses as they are maintained by different developers, should just have to run lint command at root level.

File: package.json [hunk 3]
```diff
@@ -22,13 +19,27 @@
...
+    "deploy:subgraph": "yarn workspace @quadratic-funding/subgraph run deploy"
+  },
+  "devDependencies": {
+    "@typescript-eslint/eslint-plugin": "^5.10.2",
+    "@typescript-eslint/parser": "^5.10.2",
+    "eslint": "^8.8.0",
+    "eslint-config-prettier": "^8.5.0",
+    "lerna": "^4.0.0",
+    "npm-run-all": "^4.1.5",
+    "prettier": "^2.6.0"
```
Comment (daodesigner) [inline package.json:42]: remove prettier

**Review by daodesigner — COMMENTED** [review]
Can't merge, let's hop on a call to discuss.

**Comment by daodesigner** [comment]
reopen new PR with following changes:
- [ ] do not relocate ESLint and Prettier to root project scope for consistent use
- [ ] use linter subpackage maintainers wrote lint configs for
- [ ] do not lint other's code for them 
- [ ] do not Prettier to contracts package
- [ ] Enforce single quotes to solidity files
- [ ] do not change lint settings already set up

**Comment by gurrpi** [comment]
> @gurrpi reopen new PR with following changes:
> 
> * [ ]  do not relocate ESLint and Prettier to root project scope
> * [ ]  use linter subpackage maintainers wrote lint configs for in each package
> * [ ]  do not lint other's code for them (this is generally in bad taste as it removes code attribution without actually contributing anything)
> * [ ]  do not add Prettier to contracts package
> * [ ]  Enforce single quotes to solidity files
> * [ ]  do not change lint settings already set up
> 
> In general each sub-package will be maintained by different people and there is a separation of concerns we want to enforce here, let's lean toward accepting dev conventions ( and avoid ESLint vs Prettier vs etc arguments). In the future try to not over-optimize code, if there are things you want to _change_ in the repo please open a separate issue or drop into a chat with the sub package maintainer.

I see, it was overambitious. I'll make a plan to build CI/CD pipeline and then let's make a call to discuss it.

**Pull request closed without merging**